scratch/evaluate_scheduler_qd_dense_scenario.cc

```diff
@@ -0,0 +1,825 @@
+ *
+ * Network Topology:
+ * The network consists of a single AP in the center of a room surrounded by 10 DMG STAs.
+ *
+ *
+ *                                 DMG STA (10)
+ *
+ *
+ *
+ *                  DMG STA (1)                     DMG STA (9)
+ *
+ *
+ *
+ *          DMG STA (2)                                     DMG STA (8)
+ *
+ *                                    DMG AP
+ *
+ *          DMG STA (3)                                     DMG STA (7)
+ *
+ *                                       
+ *
+ *                  DMG STA (4)                     DMG STA (6)
+ *
+ *                                     
+ *
+ *                                  DMG STA (5)
+ *
+ *
+ * Requested Service Periods:
+ * DMG STA (1) --> DMG AP
+ * DMG STA (2) --> DMG AP
+ * DMG STA (3) --> DMG AP
+ * DMG STA (4) --> DMG AP
+ * DMG STA (5) --> DMG AP
+ * DMG STA (6) --> DMG AP
+ * DMG STA (7) --> DMG AP
+ * DMG STA (8) --> DMG AP
+ * DMG STA (9) --> DMG AP
+ * DMG STA (10)--> DMG AP
+ *
+ * Running the Simulation:
+ * ./waf --run "evaluate_scheduler_qd_dense_scenario"
+ *
+ * Simulation Output:
+ * The simulation generates the following traces:
+ *
+ */
+
+NS_LOG_COMPONENT_DEFINE ("EvaluateScheduler");
+
+using namespace ns3;
+using namespace std;
+
+typedef map<Mac48Address, Ptr<Node>> Mac2NodeMap;
+typedef Mac2NodeMap::iterator Mac2NodeMapI;
+Mac2NodeMap mac2NodeMap;
+Ptr<QdPropagationLossModel> lossModelRaytracing;                   
+
+struct Parameters : public SimpleRefCount<Parameters>
+{
+  uint32_t srcNodeID;
+  Ptr<DmgWifiMac> wifiMac;
+};
+
+/* Type definitions */
+struct CommunicationPair
+{
+  Ptr<Application> srcApp;
+  Ptr<PacketSink> packetSink;
+  uint64_t totalRx = 0;
+  double throughput = 0;
+  uint64_t appDataRate;
+  Time startTime;
+};
+typedef map<Ptr<Node>, CommunicationPair> CommunicationPairList;
+typedef CommunicationPairList::iterator CommunicationPairListI;
+typedef CommunicationPairList::const_iterator CommunicationPairListCI;
+
+/** Simulation Arguments **/
+string applicationType = "onoff";             /* Type of the Tx application */
+string socketType = "ns3::UdpSocketFactory";  /* Socket Type (TCP/UDP) */
+string schedulerType = "ns3::CbapOnlyDmgWifiScheduler";   /* The type of scheduler to be used */
+string phyMode = "DMG_MCS12";                 /* The MCS to be used at the Physical Layer. */
+uint32_t packetSize = 1448;                   /* Application payload size [bytes]. */
+string tcpVariant = "NewReno";                /* TCP Variant Type. */
+uint32_t maxPackets = 0;                      /* Maximum Number of Packets */
+uint32_t msduAggregationSize = 7935;          /* The maximum aggregation size for A-MSDU [bytes]. */
+uint32_t mpduAggregationSize = 262143;        /* The maximum aggregation size for A-MPDU [bytes]. */
+double simulationTime = 10;                   /* Simulation time [s]. */
+bool csv = false;                             /* Enable CSV output. */
```

Comment:
i would remove this. looks like it's a way to print to `std::cout` in csv format and remove some logging. csv cout makes no sense (use traces instead), logging is automatically removed when running simulations is optimized build profile if `NS_LOG_UNCOND` is used in place of `std::cout`